libraries/hashed_passwords.rb

```diff
@@ -0,0 +1,44 @@
+#
+# Author:: Maksim Horbul (<max@gorbul.net>)
+# Cookbook:: mariadb
+# Library:: hashed_password
+#
+# Copyright:: 2016-2020, Eligible, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+class HashedPassword
```

Comment:
Since we also use this here:
https://github.com/sous-chefs/mariadb/blob/f661d93dd75cc73f96b35005d69fa4b256a65d4d/libraries/hashed_password.rb#L20

I wonder if we'll run into namespace issues. @tas50 @Xorima thoughts?

Reply:
We will, so need to namespace this under the cookbook name like we have in other places

Reply:
Done! The task of name spacing the mariadb class eventually was also added to my todo list